### PR TITLE
Check if there is request result #57

### PR DESCRIPTION
**Pull request by amper43**
data-kit related PR https://github.com/CorpGlory/grafana-datasource-kit/pull/58

### CODE REVIEW

File: server/src/controllers/analytics_controller.ts [hunk 1]
```diff
@@ -126,14 +126,24 @@ async function query(analyticUnit: AnalyticUnit.AnalyticUnit, detector: Analytic
...
 
-  const queryResult = await queryByMetric(
-    analyticUnit.metric,
-    panelUrl,
-    range.from,
-    range.to,
-    HASTIC_API_KEY
-  );
-  const data = queryResult.values;
+  let data;
```
Comment (rozetko) [inline server/src/controllers/analytics_controller.ts]:
```suggestion
  let data = [];
```

File: server/src/services/data_puller.ts [hunk 2]
```diff
@@ -15,6 +15,35 @@ const PULL_PERIOD_MS = 5000;
 
 export class DataPuller {
 
+  private _availableReporter = (positiveMsg: string|null, negativeMsg: string|null) => {
+    let reported = false;
+    return available => {
+      if(available && reported) {
+        reported = false;
+        if(positiveMsg) {
+         console.log(positiveMsg);
```
Comment (rozetko) [inline server/src/services/data_puller.ts]:
```suggestion
          console.log(positiveMsg);
```

File: server/src/services/data_puller.ts [hunk 3]
```diff
@@ -15,6 +15,35 @@ const PULL_PERIOD_MS = 5000;
...
+        }
+      }
+
+      if(!available && !reported) {
+        reported = true;
+        if(negativeMsg) {
+          console.log(negativeMsg);
```
Comment (rozetko) [inline server/src/services/data_puller.ts]:
```suggestion
          console.error(negativeMsg);
```

File: server/src/services/data_puller.ts [hunk 4]
```diff
@@ -15,6 +15,35 @@ const PULL_PERIOD_MS = 5000;
...
+    }
+  };
+
+  private _analyticReadyReporter = this._availableReporter(
+    'data puller: analytic ready, start pushing',
+    'data puller: analytic service not ready, return empty result'
+  );
+
+  private _grafanaConnectionRefusedReporter = this._availableReporter(
+    null,
```
Comment (rozetko) [inline server/src/services/data_puller.ts]:
```suggestion
    `data puller: connected to Grafana`,
```

File: server/src/services/data_puller.ts [hunk 5]
```diff
@@ -15,6 +15,35 @@ const PULL_PERIOD_MS = 5000;
...
+    `data puller: can't connect to Grafana. Check GRAFANA_URL.`
```
Comment (rozetko) [inline server/src/services/data_puller.ts]:
```suggestion
    'data puller: can't connect to Grafana. Check GRAFANA_URL'
```

File: server/src/services/data_puller.ts [hunk 6]
```diff
@@ -137,9 +166,21 @@ export class DataPuller {
       try {
         const time = this._unitTimes[analyticUnit.id]
         const now = Date.now();
-        return await this.pullData(analyticUnit, time, now);
+        let res = await this.pullData(analyticUnit, time, now);
```
Comment (rozetko) [inline server/src/services/data_puller.ts]:
```suggestion
        const res = await this.pullData(analyticUnit, time, now);
```

File: server/src/services/data_puller.ts (hunk 5 above)
Comment (amper43) [reply server/src/services/data_puller.ts]: invalid syntax

**Comment by rozetko** [comment]
#412

**Pull request merged**